distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/APMJvmOptions.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.server.cli;
+
+import org.elasticsearch.Build;
+import org.elasticsearch.Version;
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.common.settings.KeyStoreWrapper;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.Nullable;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * This class is responsible for working out if APM tracing is enabled, and if so, preparing
+ * a temporary config file for the APM Java agent and CLI options to the JVM to configure APM.
+ */
+class APMJvmOptions {
+    /**
+     * Contains agent configuration that must always be applied, and cannot be overridden.
+     */
+    private static final Map<String, String> STATIC_CONFIG;
+
+    /**
+     * Contains default configuration that will be used unless overridden by explicit configuration.
+     */
+    private static final Map<String, String> CONFIG_DEFAULTS;
+
+    /**
+     * Lists all APM configuration keys that are not dynamic and must be configured via the config file.
+     */
+    private static final List<String> STATIC_AGENT_KEYS = List.of(
+        "api_key",
+        "aws_lambda_handler",
+        "breakdown_metrics",
+        "classes_excluded_from_instrumentation",
+        "cloud_provider",
+        "data_flush_timeout",
+        "disable_metrics",
+        "disable_send",
+        "enabled",
+        "enable_public_api_annotation_inheritance",
+        "environment",
+        "global_labels",
+        "hostname",
+        "include_process_args",
+        "log_ecs_formatter_allow_list",
+        "log_ecs_reformatting_additional_fields",
+        "log_ecs_reformatting_dir",
+        "log_file",
+        "log_file_size",
+        "log_format_file",
+        "log_format_sout",
+        "max_queue_size",
+        "metrics_interval",
+        "plugins_dir",
+        "profiling_inferred_spans_lib_directory",
+        "secret_token",
+        "service_name",
+        "service_node_name",
+        "service_version",
+        "stress_monitoring_interval",
+        "trace_methods_duration_threshold",
+        "use_jaxrs_path_as_transaction_name",
+        "verify_server_cert"
+    );
+
+    static {
+        STATIC_CONFIG = new HashMap<>();
```

Comment:
It's always a little weird to have mutable constants, maybe better use `Map.of(` or `Map.copyOf` for the assignment of the constants here?

Reply:
I was trying to avoid the formatting getting mangled, but actually all I need to do is use the right enable / disable comments. Now we just have `Map.of()`.